t/semaphore.t

```diff
@@ -0,0 +1,1068 @@
+_EOC_
+
+our $HttpConfigInitByLua= <<_EOC_;
+    lua_package_path "$pwd/lib/?.lua;;";
+    init_by_lua '
+            local g = _G
+            local semaphore = require "ngx.semaphore"
+            local sem, err = semaphore.new(0)
+            g.err = err
+    ';
+_EOC_
+
+our $HttpConfigIntWorkerBy = <<_EOC_;
+    lua_package_path "$pwd/lib/?.lua;;";
+    init_worker_by_lua '
+            local semaphore = require "ngx.semaphore"
+            local sem = semaphore.new(0)
+            local ok, err = sem:wait(1)
+            local g = _G
+            g.test = err
+    ';
+
+_EOC_
+
+run_tests();
+
+__DATA__
+
+=== TEST 1: basic semaphore wait post
+--- http_config eval: $::HttpConfig
+--- config
+    location = /test {
+        access_log off;
+        content_by_lua '
+        local res1, res2 = ngx.location.capture_multi{
+          { "/sem_wait"},
+          { "/sem_post"},
+        }
+        ngx.say(res1.status)
+        ngx.say(res1.body)
+        ngx.say(res2.status)
+        ngx.say(res2.body)
+        ';
+    }
+    location /sem_wait {
+        content_by_lua '
+            local semaphore = require "ngx.semaphore"
+            local g = getmetatable(_G).__index
+
+            if not g.test then
+                local sem, err = semaphore.new(0)
+                if not sem then
+                    ngx.say(err)
+                    return
+                end
+                g.test = sem
+            end
+            local sem = g.test
+            local ok, err = sem:wait(1)
+            if ok then
+                ngx.print("wait")
+            end
+        ';
+    }
+
+    location /sem_post {
+        content_by_lua '
+            local semaphore = require "ngx.semaphore"
+            local g = getmetatable(_G).__index
+            if not g.test then
+                local sem, err = semaphore.new(0)
+                if not sem then
+                    ngx.say(err)
+                    ngx.exit(500)
+                end
+                g.test = sem
+            end
+            local sem = g.test
+            ngx.sleep(0.001)
+            collectgarbage("collect")
+            local ok, err = sem:post()
+            if ok then
+                ngx.print("post")
+                ngx.exit(200)
+            else
+                ngx.exit(500)
+            end
+
+    ';
+    }
+--- no_check_leak
+--- request
+GET /test
+--- response_body
+200
+wait
+200
+post
+--- no_error_log
+[error]
+[crit]
+
+
+
+=== TEST 2: basic semaphore wait post
+--- http_config eval: $::HttpConfig
+--- config
+    location /test {
+        content_by_lua '
+            local semaphore = require "ngx.semaphore"
+            local sem = semaphore.new(0)
+            local ok, err = sem:wait(0.1)
+            ngx.say(tostring(ok).." "..tostring(err))
+        ';
+    }
+--- no_check_leak
+--- request
+GET /test
+--- response_body
+false timeout
+--- no_error_log
+[error]
+
+
+
+=== TEST 3: basic semaphore wait not allow in body_filter_by_lua
```

Comment:
Grammar: "not allow in" -> "not allowed in". Please fix the test titles in subsequent test blocks as well.